src/Engine/GameAction.java

```diff
@@ -80,13 +81,117 @@ public GameEventQueue getEvents(MapMaster gameMap)
+  /** Similar to UnitProductionAction, except occupancy/financial prerequisites are waived. */
+  public static class UnitSpawnAction extends GameAction
+  {
+    private final XYCoord where;
+    private final Commander who;
+    private final UnitModel what;
+    private final CreateUnitEvent.AnimationStyle how;
+    private final boolean allowStomping;
+    private final boolean unitIsReady;
+
+    /**
+     * Creates a new `what` unit at location `where`, belonging to `who`. No funds will be taken, and unit creation will fail
+     * if there is already another unit at `where`.
+     */
+    public UnitSpawnAction(Commander who, UnitModel what, XYCoord where)
+    {
+      this(who, what, where, CreateUnitEvent.AnimationStyle.NONE, false, false);
+    }
+    /**
+     * Creates a new `what` unit at location `where`, belonging to `who`. No funds will be taken. If `allowStomping` and a unit
+     * is already present at `where`, the existing unit will be killed. If `unitIsReady`, then the new unit will be able to move immediately.
+     */
+    public UnitSpawnAction(Commander who, UnitModel what, XYCoord where, CreateUnitEvent.AnimationStyle how, boolean allowStomping, boolean unitIsReady)
+    {
+      this.where = where;
+      this.who = who;
+      this.what = what;
+      this.how = how;
+      this.allowStomping = allowStomping;
+      this.unitIsReady = unitIsReady;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // UNITPRODUCTION actions consist of
+      //   CREATEUNIT
+      //   [MASSDAMAGE] (if an existing unit is stomped/killed by the new unit)
+      //   [DEATH]      (if an existing unit is stomped/killed by the new unit)
+      //   [DEFEAT]     (if a unit getting stomped is the last one for that CO)
+      GameEventQueue buildEvents = new GameEventQueue();
+
+      // Validate events.
+      boolean isValid = true;
+      isValid &= (null != gameMap) && (null != who) && (null != what) && (null != where) && gameMap.isLocationValid(where)
+          && (allowStomping || gameMap.isLocationEmpty(where));
+      if( isValid )
+      {
+        buildEvents.add(new CreateUnitEvent(who, what, where, how, unitIsReady, allowStomping));
+
+        // Figure out if something's in the way; if so, deal with it.
+        Unit obstacle = gameMap.getLocation(where).getResident();
+        if( null != obstacle )
+        {
+          ArrayList<Unit> ary = new ArrayList<Unit>();
+          ary.add(obstacle);
+          boolean fatal = true;
+          MassDamageEvent mde = new MassDamageEvent(who, ary, obstacle.getHP()+1, fatal);
+          UnitDieEvent ude = new UnitDieEvent(obstacle);
+          buildEvents.add(mde);
+          buildEvents.add(ude);
+
+          // Poor sap died; Check if his CO lost the game.
+          if( obstacle.CO.units.size() == 1 )
+          {
+            CommanderDefeatEvent cde = new CommanderDefeatEvent(obstacle.CO);
+            buildEvents.add(cde);
+          }
```

Comment:
This could create a situation where you stomp your *own* last unit and lose even though before and after the stomp you had 1 unit left.
Could go either way on that, lol.